ec/src/msm/fixed_base.rs

```diff
@@ -81,17 +81,17 @@ impl FixedBase {
 
     // TODO use const-generics for the scalar size and window
     // TODO use iterators of iterators of T::Affine instead of taking owned Vec
-    pub fn msm<T: ProjectiveCurve>(
-        scalar_size: usize,
-        window: usize,
-        table: &[Vec<T::Affine>],
-        v: &[T::ScalarField],
-    ) -> Vec<T> {
+    pub fn msm<C, I>(scalar_size: usize, window: usize, table: &[Vec<C::Affine>], v: I) -> Vec<C>
+    where
+        C: ProjectiveCurve,
+        I: IntoIterator,
+        I::Item: Borrow<C::ScalarField>,
```

Comment:
Does this more generic bound but us much? the `IntoIterator` I can understand (but we will want to fix it to be conditionally `IntoParallelIterator`, or call `par_bridge()` on it.)

However, the `Borrow` one doesn't seem as useful, IMO

Reply:
mmmmh, I was trying to emulate @oleganza  https://github.com/dalek-cryptography/curve25519-dalek/blob/main/src/backend/serial/scalar_mul/pippenger.rs. 
Looking at https://github.com/search?q=optional_multiscalar_mul&type=code there's a bunch of repos that use `.iterator().chain()` on vectors (which will lead to iterators of type `&C::ScalarField`). I'm however happy to move this into a separate pull request.

Reply:
@mmagician, @Pratyush: with a ```% git checkout master -- ec/src/msm/fixed_base.rs ``` we are all good with the new `VariableBaseMSM` API, only the Changelog might need minor edits.

Reply:
The impl in curve25519-dalek is not parallelized, which is why they're fine with `Iterator`. In our case, since we have parallelization, and because iterators are lazy, if you do `iter().chain()`, the `chain()` call will be evalauted many times. For more complex iterators this can be an issue.

Reply:
This response is not related to Borrow, is it?

Reply:
@mmaker So is the suggestion to provide an Iterator-based API for MSMs with Scalars, while providing the standard slice-based API for MSMs with bigints?